File: backend/app/schemas/gmail.py
from pydantic import AliasChoices, BaseModel, ConfigDict, Field


class GmailRecentEmail(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    gmail_id: str
    thread_id: str
    from_: str = Field(
        validation_alias=AliasChoices("from", "sender"),
        serialization_alias="from",
    )
    recipients: str = ""
    subject: str = ""
    date: str = ""
    snippet: str = ""
    body_text: str = ""
    has_attachment: bool = False
    internal_date_ms: int = 0


class GmailRecentResponse(BaseModel):
    source: str = "gmail"
    count: int
    emails: list[GmailRecentEmail]

File: frontend/src/lib/types/gmail.ts
export interface GmailRecentEmail {
  gmail_id: string;
  thread_id: string;
  from: string;
  recipients?: string;
  subject: string;
  date: string;
  snippet: string;
  body_text?: string;
  has_attachment?: boolean;
  internal_date_ms?: number;
}

/** Raw API payload may use alternate field names before normalization. */
export interface GmailRecentEmailRaw {
  gmail_id?: string;
  thread_id?: string;
  from?: string;
  sender?: string;
  recipients?: string;
  subject?: string;
  date?: string;
  snippet?: string;
  body_text?: string;
  has_attachment?: boolean;
  internal_date_ms?: number;
}

export interface GmailRecentResponse {
  source: string;
  count: number;
  emails: GmailRecentEmailRaw[];
}
